nemo/collections/asr/parts/submodules/rnnt_beam_decoding.py

```diff
@@ -1319,21 +1365,47 @@ def prefix_search(
                         self.joint.joint(enc_out, hyp_i.dec_out[-1]) / self.softmax_temperature, dim=-1,
                     )
                     logp = logp[0, 0, 0, :]
-
                     curr_score = hyp_i.score + float(logp[hyp_j.y_sequence[pref_id]])
+                    # Setup ngram LM:
+                    if self.ngram_lm:
+                        lm_score, next_state = self.compute_ngram_score(
+                            hyp_i.ngram_lm_state, int(hyp_j.y_sequence[pref_id])
+                        )
+                        curr_score += self.ngram_lm_alpha * lm_score
 
                     for k in range(pref_id, (curr_id - 1)):
                         logp = torch.log_softmax(
                             self.joint.joint(enc_out, hyp_j.dec_out[k]) / self.softmax_temperature, dim=-1,
                         )
                         logp = logp[0, 0, 0, :]
-
                         curr_score += float(logp[hyp_j.y_sequence[k + 1]])
+                        # Setup ngram LM:
+                        if self.ngram_lm:
+                            lm_score, next_state = self.compute_ngram_score(next_state, int(hyp_j.y_sequence[k + 1]))
+                            curr_score += self.ngram_lm_alpha * lm_score
 
                     hyp_j.score = np.logaddexp(hyp_j.score, curr_score)
 
         return hypotheses
 
+    def compute_ngram_score(self, current_lm_state: "kenlm.State", label: int) -> Tuple[float, "kenlm.State"]:
+        """
+        Score computation for kenlm ngram language model.
+        """
+
+        next_state = kenlm.State()
+        lm_score = self.ngram_lm.BaseScore(current_lm_state, chr(label + self.token_offset), next_state)
+        lm_score *= 1.0 / np.log10(np.e)
+
+        return lm_score, next_state
+
+    def set_decoding_type(self, decoding_type: str):
+
+        # Please check train_kenlm.py in scripts/asr_language_modeling/ to find out why we need
+        # TOKEN_OFFSET for BPE-based models
+        if decoding_type == 'subword':
+            self.token_offset = 100
```

Comment:
Let's define this number in one of the files and then import it here.

Reply:
Good suggestion! I imported token_offset variable from train_kenlm.py script.

Reply:
No, import from train_kenlm.py was bad idea. Do you have any suggestion where in nemo I can add default `token_offset` value? I can create `nemo/collections/asr/parts/submodules/decoder_constants.py` with this parameter.

BTW: [CTC beam search ](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/parts/submodules/ctc_beam_decoding.py#L559) also has hardcoded value for `token_offset`.

Reply:
Keep the offset inside of the CTC beam decoding config. The offset is not used for RNNT. Import the constant from the Nemo CTC beam search decoding file inside of train kenlm file

Reply:
Ok, I will add offset value inside `ctc_beam_decodding.py`. The idea is to use the same n-gram lm model for CTC_bpe and RNNT_bpe models. That is why I want to add offset for RNNT_bpe model too.